app/workers/process_docker_image_worker.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+require 'minitar'
+require 'zlib'
+
+class ProcessDockerImageWorker < BaseWorker
+  MIN_TARBALL_SIZE  = 5.bytes    # to avoid processing empty or invalid tarballs
+  MAX_TARBALL_SIZE  = 1.gigabyte # to avoid downloading excessive tarballs
+  MAX_MANIFEST_SIZE = 1.megabyte # to avoid storing large manifests
+
+  sidekiq_options queue: :critical
+
+  def perform(artifact_id)
+    artifact = ReleaseArtifact.find(artifact_id)
+    return unless
+      artifact.processing?
+
+    # sanity check the image tarball
+    unless artifact.content_length.in?(MIN_TARBALL_SIZE..MAX_TARBALL_SIZE)
+      raise ImageNotAcceptableError, 'unacceptable filesize'
+    end
+
+    # download the image tarball in chunks to reduce memory footprint
+    client = artifact.client
+    enum   = Enumerator.new do |yielder|
+      client.get_object(bucket: artifact.bucket, key: artifact.key) do |chunk|
+        yielder << chunk
+      end
+    end
+
+    # wrap the enumerator to provide an IO-like interface
+    tar = EnumeratorIO.new(enum)
+
+    # keep a ref to the manifest
+    manifest = nil
+
+    # unpack the package tarball
+    unpack tar do |archive|
+      archive.each do |entry|
+        case entry.name
+        in 'index.json'
+          # TODO(ezekg) do we need this for OCI compliance or can we generate dynamically?
```

Comment:
This is required for multi-platform images. Clients can request it using the `application/vnd.docker.distribution.manifest.list.v2+json` content type, vs the `application/vnd.docker.distribution.manifest.v2+json` content type.

Reply:
I think we can actually use this in place of `manifest.json`, since  it's Docker-specific and seems `index.json` replaces it, at least for the OCI.